packages/compass-connections-navigation/src/connections-navigation-tree.tsx

```diff
@@ -378,68 +390,52 @@ const ConnectionsNavigationTree: React.FunctionComponent<
   // eslint-disable-next-line @typescript-eslint/ban-ts-comment
   // @ts-expect-error ignoring test props so they are not part of the interface
   __TEST_REACT_WINDOW_OVERSCAN = null,
-  ...restProps
 }) => {
   // we'll have either connections for MC version, or databases for SC version
-  const {
-    isSingleConnection,
-    data,
-  }: {
-    isSingleConnection: boolean;
-    data:
-      | MCConnectionsNavigationTreeProps['connections']
-      | SCConnectionsNavigationTreeProps['databases'];
-  } = useMemo(() => {
-    if ((restProps as SCConnectionsNavigationTreeProps).databases) {
-      return {
-        isSingleConnection: true,
-        data: (restProps as SCConnectionsNavigationTreeProps).databases,
-      };
-    }
-    return {
-      isSingleConnection: false,
-      data: (restProps as MCConnectionsNavigationTreeProps).connections,
-    };
-  }, [restProps]);
+  const isSingleConnection = !usePreference(
+    'enableNewMultipleConnectionSystem'
+  );
 
   const listRef = useRef<List | null>(null);
   const id = useId();
 
   useEffect(() => {
     if (activeNamespace) {
-      onDatabaseExpand(activeNamespace, true);
+      for (const connection of connections) {
+        onDatabaseExpand(connection.connectionInfo.id, activeNamespace, true);
+      }
     }
-  }, [activeNamespace, onDatabaseExpand]);
+  }, [activeNamespace, onDatabaseExpand, connections]);
   // TODO(COMPASS-7775): the we'll need something similar to expand the active connection
 
   const items: TreeItem[] = useMemo(() => {
     if (!isSingleConnection) {
-      return (data as MCConnectionsNavigationTreeProps['connections']).flatMap(
-        (connection, connectionIndex) =>
-          connectionToItems({
-            connection,
-            connectionIndex,
-            connectionsLength: data.length,
-            expanded: expanded as MCConnectionsNavigationTreeProps['expanded'],
-          })
+      return connections.flatMap((connection, connectionIndex) =>
+        connectionToItems({
+          connection,
+          connectionIndex,
+          connectionsLength: connections.length,
+          expanded: expanded,
+        })
       );
     } else {
-      return (data as SCConnectionsNavigationTreeProps['databases']).flatMap(
-        (database, databaseIndex) =>
-          databaseToItems({
-            database,
-            databaseIndex,
-            databasesLength: data.length || 0,
-            expanded: expanded as SCConnectionsNavigationTreeProps['expanded'],
-            level: 1,
-          })
+      const connection = connections[0];
+      return connection.databases.flatMap((database, databaseIndex) =>
+        databaseToItems({
+          connectionId: connection.connectionInfo.id, // 'TOFIX'
```

Comment:
what's wrong with this?

Reply:
Nothing anymore, I had this marker because the connectionInfo was hardcoded during development :D.